chapter9/phone-agent/speech.py

```
@@ -0,0 +1,307 @@
+        )
+        if not self.python:
+            raise RuntimeError(
+                "local ASR requires torch and openai-whisper; set WHISPER_PYTHON to that Python executable"
+            )
+        self.model = os.getenv("WHISPER_MODEL", "tiny")
+
+    @staticmethod
+    def _available(python: str) -> bool:
+        try:
+            return (
+                subprocess.run(
```

Ruff (0.16.0)

[error] `subprocess` call: check for execution of untrusted input

(S603)